src/test/java/com/google/jenkins/plugins/storage/IT/ClassicUploadStepPipelineIT.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jenkins.plugins.storage.IT;
+
+import static com.google.jenkins.plugins.storage.IT.ITUtil.deleteFromBucket;
+import static com.google.jenkins.plugins.storage.IT.ITUtil.dumpLog;
+import static com.google.jenkins.plugins.storage.IT.ITUtil.formatRandomName;
+import static com.google.jenkins.plugins.storage.IT.ITUtil.loadResource;
+import static org.junit.Assert.assertNotNull;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.google.jenkins.plugins.credentials.oauth.GoogleRobotPrivateKeyCredentials;
+import com.google.jenkins.plugins.credentials.oauth.ServiceAccountConfig;
+import com.google.jenkins.plugins.storage.StringJsonServiceAccountConfig;
+import hudson.EnvVars;
+import hudson.model.Result;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
+import java.util.logging.Logger;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ClassicUploadStepPipelineIT {
```

Comment:
Suggest also adding a test to cover the case when the pipeline syntax is valid, but the bucket doesn't exist.

Reply:
Adding a non-existent bucket leads to a new bucket getting created.
You can see that here:
https://github.com/jenkinsci/google-storage-plugin/blob/631adc9c0724bd404012277cfe9e38087d8eb40f/src/main/java/com/google/jenkins/plugins/storage/AbstractUpload.java#L572-L577

So it doesn't lead to a failure, just leads to a new bucket getting created.

Reply:
Understood. Thanks :)